src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -74,7 +74,10 @@ public Task<VisualStudioProject> CreateAndAddToWorkspaceAsync(string projectSyst
                 ? filePath
                 : null;
 
-            await _visualStudioWorkspaceImpl.EnsureDocumentOptionProvidersInitializedAsync(cancellationToken).ConfigureAwait(true);
+            // After the call to EnsureDocumentOptionProvidersInitializedAsync, everything can be off the UI thread.
+            // Thus, we have a ConfigureAwait(false) on the call and switch explicitly after.
+            await _visualStudioWorkspaceImpl.EnsureDocumentOptionProvidersInitializedAsync(cancellationToken).ConfigureAwait(false);
+            await TaskScheduler.Default;
```

Comment:
Don't `ConfigureAwait(false)` and `await TaskScheduler.Default` do the same thing?

It's possible `await TaskScheduler.Default` is already completed so it runs synchronously, but seems redundant :D

Reply:
@Therzok: the .ConfigureAwait(false) only matters if the EnsureDocumentOptionProvidersInitializedAsync actually yields; if it completes synchronously we'd stay on the UI thread.

Reply:
Which to clarify: that means we need the await TaskScheduler.Default; but as @sharwell pointed out we don't want to have the first await come to the UI thread only to immediately bounce to the UI thread, so we kinda end up with this (admittedly odd) pattern.

Reply:
That is very good to know, thank you!